### PR TITLE
Selenium_Assignment1

### PR DESCRIPTION
**Pull request by PallaviSTechnocredits**

### CODE REVIEW

File: seleniumByTechnocredits/src/seleniumPractise/pallavi/Login_Password_Validation.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		Thread.sleep(3000);
+		boolean flag = driver.findElement(By.xpath(".//*[text()='Login']")).isDisplayed();
+
+		if (flag == true) 
+		{
+			driver.findElement(By.xpath(".//input[@id='unameSignin' and @class ='form-control' ]")).sendKeys("Pallavi");
+			String password = "ab@123";
+			driver.findElement(By.xpath(".//input[@id='pwdSignin' and @class='form-control']")).sendKeys(password);
+			driver.findElement(By.xpath(".//button[@id='btnsubmitdetails']")).click();
+			if (password.length() >= 5 && password.length() <= 8)
```
Comment (PrachiJTechnocredits) [inline seleniumByTechnocredits/src/seleniumPractise/pallavi/Login_Password_Validation.java]: Pallavi here validation of password length is only for less than equal to 8. 
As if password length is less than 5 then also is should show error.

File: seleniumByTechnocredits/src/seleniumPractise/pallavi/Alert_Verification.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+	public static void main(String[] args) throws InterruptedException 
+	{
+		WebDriver driver =Init.initChromeDriver();
+		driver.get("http://automationbykrishna.com/#");
+		driver.manage().window().maximize();
+		driver.findElement(By.linkText("Basic Elements")).click();
+		Thread.sleep(1000);
+		String expectedResult = "You must be TechnoCredits student!!";
+		driver.findElement(By.xpath(".//button[text()='Alert']")).click();
```
Comment (PrachiJainTR) [inline seleniumByTechnocredits/src/seleniumPractise/pallavi/Alert_Verification.java]: Hi Pallavi, Its giving below error when I ran your code:
Exception in thread "main" org.openqa.selenium.WebDriverException: unknown error: Element <button type="submit" class="btn btn-success" id="javascript..." onclick="callJavaScriptAlert()">Alert</button> is not clickable at point (73, 580).

You can write the code to scroll till element before clicking.
Same happened for next two programs.

File: seleniumByTechnocredits/src/seleniumPractise/pallavi/Registration_Validation.java [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+		aName.clear();
+		aName.sendKeys(address);
+		WebElement eName = driver.findElement(By.id("useremail"));
+		eName.sendKeys(email);
+		scroll();
+		ele.click();
+
+		Alert alt = driver.switchTo().alert();
+		Alert alt1 = driver.switchTo().alert();
+		Alert alt2 = driver.switchTo().alert();
```
Comment (PrachiJainTR) [inline seleniumByTechnocredits/src/seleniumPractise/pallavi/Registration_Validation.java]: Please check once alt1 and alt2 is required or not.

File: seleniumByTechnocredits/src/seleniumPractise/pallavi/Login_Password_Validation.java (hunk 1 above)
Comment (AnkitaGTechnocredits) [inline seleniumByTechnocredits/src/seleniumPractise/pallavi/Login_Password_Validation.java]: Hi Pallavi,
Please consider below review comment:
1. Alert is also generating on password length less than 5. So the condition to match password length should only be less than or equal to 8.

File: seleniumByTechnocredits/src/seleniumPractise/pallavi/Registration_Validation.java (hunk 3 above)
Comment (AnkitaGTechnocredits) [inline seleniumByTechnocredits/src/seleniumPractise/pallavi/Registration_Validation.java]: Hi Pallavi,

Also adding to Prachi's  review comment, alt2 object of alert is not used to validate any alert message.

File: seleniumByTechnocredits/src/seleniumPractise/pallavi/Registration_Validation.java [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+		uName.sendKeys(userName);
+		ele.click();
+		if (userName.length() <= 5) 
+		{
+			Alert al = driver.switchTo().alert();
+			Thread.sleep(1000);
+			String alertMessage = al.getText();
+			Thread.sleep(1000);
+			al.accept();
+			String message = "Username length should be greater then 5 characters.";
```
Comment (AnkitaGTechnocredits) [inline seleniumByTechnocredits/src/seleniumPractise/pallavi/Registration_Validation.java:271]: Hi Pallavi,

I believe after validating alert message then alert should get closed. So in If block of message validation accepting alert code should be written.

**Comment by PallaviSTechnocredits** [comment]
updated with reviewer comments

**Pull request closed without merging**